core/src/bmi.jl

```diff
@@ -59,6 +59,12 @@ function BMI.initialize(T::Type{Model}, config::Config)::Model
     return Model(integrator, config, saved_flow)
 end
 
+function BMI.finalize(model::Model)
+    write_basin_output(model)
+    write_flow_output(model)
+    write_control_output(model)
+end
```

Comment:
Let's return model just like the other BMI functions, rather than whatever `write_control_output` returns.

```suggestion
function BMI.finalize(model::Model)::Model
    write_basin_output(model)
    write_flow_output(model)
    write_control_output(model)
    return model
end
```

Reply:
Sounds good!